File: examples/clj/test/com/rpl/agent/basic/dataset_agent_test.clj
(ns com.rpl.agent.basic.dataset-agent-test
  (:require
   [clojure.test :refer [deftest testing is]]
   [com.rpl.agent-o-rama :as aor]
   [com.rpl.rama :as rama]
   [com.rpl.rama.test :as rtest]
   [com.rpl.agent.basic.dataset-agent :refer [DatasetExampleModule math-input-schema math-output-schema]]
   [clojure.string :as str]))

(deftest dataset-agent-test
  ;; Test that the dataset agent runs without errors
  (testing "dataset-agent"
    (testing "runs without errors"
      (is (nil? (com.rpl.agent.basic.dataset-agent/-main))
          "Main function should complete without throwing exceptions"))))

File: examples/java/basic/src/test/java/com/rpl/agent/basic/DatasetAgentTest.java
package com.rpl.agent.basic;

import org.junit.Test;

/**
 * Test class for DatasetAgent demonstrating dataset lifecycle management testing.
 *
 * <p>This test demonstrates:
 *
 * <ul>
 *   <li>Testing that the dataset agent main method runs without errors
 *   <li>Verifying dataset lifecycle management operations execute successfully
 * </ul>
 */
public class DatasetAgentTest {

  @Test
  public void testDatasetAgent() throws Exception {
    // Test that the dataset agent runs without errors
    DatasetAgent.main(new String[] {});
  }
}
